File: solution/120/solution.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        if(triangle.size()==0)return 0;
        vector<vector<int> > result;
        for(int i=0;i<triangle.size();i++){
            vector<int> onelist;
            for(int j=0;j<triangle[i].size();j++){
                onelist.push_back(0);
            }
            result.push_back(onelist);
        }
        result[triangle.size()-1] = triangle[triangle.size()-1];
        for(int i = triangle.size()-2;i>=0;i--){
            for(int j=0;j<triangle[i].size();j++){
                result[i][j] = triangle[i][j] + min(result[i+1][j], result[i+1][j+1]);
            }
        }
        return result[0][0];
    }
};

File: solution/120/solution.py
class Solution(object):
    def minimumTotal(self, triangle):
        """
        :type triangle: List[List[int]]
        :rtype: int
        """
        if len(triangle) == 0:
            return 0
        result = [[] for x in xrange(len(triangle))]
        result[-1] = triangle[-1]
        for line in xrange(len(triangle)-2, -1, -1):
            for i in xrange(len(triangle[line])):
                result[line].append(min(result[line+1][i], result[line+1][i+1]) + triangle[line][i])
        return result[0][0]